gigascience/gigadb-website
# Migration script for transferring datasets from CNGB backup server to Wasabi

# Pull request for issue: #1130

This is a pull request for the following functionalities:

* Batch copy of GigaDB dataset files to `live` directory in Wasabi `gigadb-website` bucket
* Default test mode that copies files to `gigadb-website/dev` and `gigadb-website/staging`bucket directories.
* Logging of status messages
* Docker deployment of dataset migration tool on CNGB backup server

## How to test?

See [README.md](https://github.com/pli888/gigadb-website/blob/rclone-cngb2wasabi/gigadb/app/tools/wasabi-migration/README.md) for documentation on how to test and use this tool to migrate GigaDB data.

## How have functionalities been implemented?

The main functionality for copying dataset files to Wasabi is provided by the `rclone_copy.sh`. This script is deployed with the `rclone` tool in a Docker container which comes with test data to perform test dataset uploads to the Wasabi `gigadb-datasets` bucket and also has access to live data when installed on the CNGB backup server.

## Any issues with implementation?

Configuration to use rclone on the CNGB backup server needs to be generated on your local machine.

## Any changes to documentation?

Information on how to use this Wasabi migration tool is provided by the [README.md](https://github.com/pli888/gigadb-website/blob/rclone-cngb2wasabi/gigadb/app/tools/wasabi-migration/README.md) file.
